File: openai-api/dall-e.js
const openai = require("openai");
if (!process.env.OPENAI_API_KEY) {
  throw Error("OpenAI API Key Not Loaded");
}
const clientOpenAI = new openai.OpenAI({ apiKey: process.env.OPENAI_API_KEY });

const generateImage = async (prompt) => {
  const res = await clientOpenAI.images.generate({
    prompt,
    model: "dall-e-3",
    n: 1,
    size: "1024x1024",
  });
  return res?.data?.[0].url;
};

const main = async () => {
	console.log(await generateImage("logo for a website"));
};

main();

File: openai-api/dall-e.py
import os
from openai import OpenAI

if not os.getenv("OPENAI_API_KEY"):
    raise ValueError("OpenAI API Key Not Loaded")

client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))

def generate_image(prompt):
    response = client.images.generate(
        model="dall-e-3",
        prompt=prompt,
        n=1,
        size="1024x1024",
    )
    return response.data[0].url

if __name__ == "__main__":
    print(generate_image("logo for a website"))
